NethermindNode.Tests/Tests/Pruning/JsonRpcPruning.cs

```diff
@@ -69,31 +76,43 @@ public void ShouldPruneDbUsingAdminRpc()
                 "Full Pruning Finished"
             };
 
-            int expectedLogIndex = 0;
+            HashSet<string> missingLogs = new HashSet<string>(expectedLogs);
 
             try
             {
-                foreach (var line in DockerCommands.GetDockerLogs("sedge-execution-client", "Full Pruning", true, cts.Token))
+                foreach (var line in DockerCommands.GetDockerLogs("sedge-execution-client", "Full Pruning", true, cts.Token, "--since 2m")) //since to ensure that we will get only recent logs but including all from beggining of test
                 {
-                    Console.WriteLine(line);
-
-                    if (line.Contains(expectedLogs[expectedLogIndex]))
-                    {
-                        expectedLogIndex++;
-                    }
+                    Console.WriteLine(line); // For visibility during testing
 
-                    if (expectedLogIndex >= expectedLogs.Length)
+                    foreach (var expectedLog in missingLogs)
                     {
-                        break;
+                        if (line.Contains(expectedLog))
+                        {
+                            Logger.Info($"Log found: \"{line}\" - Expected log: {expectedLog}");
+                            missingLogs.Remove(expectedLog);
+
+                            if (expectedLog == expectedLogs.Last())
+                            {
+                                // End becuase Pruning itself may work but for some reason some logs may not be found - so better this way than waiting for all
+                                cts.Cancel();
+                                break;
+                            }
+                        }
```

Comment:
```suggestion
                        if (line.Contains(expectedLog))
                        {
                            break;
                        }
                        
                        Logger.Info($"Log found: \"{line}\" - Expected log: {expectedLog}");
                        missingLogs.Remove(expectedLog);
  
                        if (expectedLog == expectedLogs.Last())
                        {
                            // End because Pruning itself may work but for some reason some logs may not be found - so better this way than waiting for all
                            cts.Cancel();
                            break;
                        }
```

Reply:
if (!line.Contains(expectedLog))
                        {
                            break;
                        }
                        
This if would exit entire foreach so it would always exit on very first log items and did not check others which are supposed to be there.

Reply:
oh, right. I actually mean to use `continue`. The general idea is to use a guard clause, to eliminate 1 level of nesting. Not required, but makes it easier to read/follow.

Reply:
OK with continue applied - agree that would be better to have 1 less nesting